Add visualization and other rework

penrods opened the pull request:
Changes include:
* Make all text translatable
* Have eyes show timer progress
* Unify duration formatting (spoken and visual)
* Add "Mute timer" command
* Show flashing overtime message when timer expires
* Rework of internal active_timer structure
* Disable 'Cancel', 'Status' and 'Mute' intents when there is no running timer
* Switch from convers() to get_response() implementation for multiple timers
* Removing unused libraries

Commit:
Add visualization and other rework

Changes include:
* Make all text translatable
* Have eyes show timer progress
* Unify duration formatting (spoken and visual)
* Add "Mute timer" command
* Show flashing overtime message when timer expires
* Rework of internal active_timer structure
* Disable 'Cancel', 'Status' and 'Mute' intents when there is no running timer
* Switch from convers() to get_response() implementation for multiple timers
* Removing unused libraries

penrods commented:
This does have some TODOs, the biggest being that it no longer handles mixed units when creating a timer.   This should be solved in a general parser within mycroft.core, however.

LearnedVector commented on __init__.py:
```diff
@@ -12,314 +12,420 @@
+    # TODO: Doesn't handle 'start 1 and a half minute timer'
+    @intent_file_handler('start.timer.intent')
+    def handle_start_timer(self, message):
         duration = message.data["duration"]
```
errors out if the duration entity is not found. Having a get response to ask for the duration would be good here.

LearnedVector commented on __init__.py:
```diff
@@ -12,314 +12,420 @@
+    def handle_mute_timer(self, message):
+        self.mute = True
 
+    @intent_handler(IntentBuilder("").require("Cancel").require("Timer").optionally("All"))
```
cancel timer doesn't work by voice. The intent may not be registering?

LearnedVector replied:
Also if multiple timers are set, and if you hit the top button, Mycroft will ask N times consecutively if you'd like to cancel the timer. N is the numbers of timers set.

LearnedVector commented on __init__.py:
```diff
@@ -12,314 +12,420 @@
+
+    # Handles 'How much time left'
+    @intent_file_handler('status.timer.intent')
+    def handle_status_timer(self, message):
```
when setting multiple timers, mycroft will only give you status of the most recently set timer, and not the others even when you explicitly ask for it. It seems to hit the handle_stat_timer intent though nothing is spoken.

LearnedVector replied:
I've reset the skills service just in case something was wonky on load. It seems that asking for "status of the timer", then explicitly say "5 minute timer" afterward works, but saying "status of the 5 minute timer" does not if you set another timer after the 5 minute timer.

penrods replied:
Release 0.9.17 broke several things.  There is a pending mycroft-core fix for this, but I think I have it working for now.  Will need to remove the "hack" after next release.

LearnedVector reviewed (changes requested):
just a few quirks that i saw while testing. This could still be merged and have Dereck add to the UX notes of what he observes

Commit:
Hack around enable_intent() bug and tweaks

Build 0.9.17 broke the enable_intent() method.  This has been corrected and
will bein the next release, but in the meantime this change works around the
bug.

Also:
* Normalize the timer name based on a consistent formatting of the time.
* Prototype using a different display
* Singular/plural response was backwards when cancelling timer(s)

Commit:
Add support for labels

Support timers with labels and allow referring to a timer by:
* Name
* Original duration (normalized)
* Index

LearnedVector reviewed (approved):
LGTM
